producers/src/end-to-end-test/java/uk/gov/moj/sdt/producers/SubmitBulkPerformanceTest.java

```diff
@@ -1,312 +1,318 @@
+    }
+
+    /**
+     * @param requestType the individual request type.
+     */
+    private void scheduleRequestToProcess(final BulkRequestType requestType) {
+        Thread thread = new Thread(new BulkRequestProcessor(Thread.currentThread(), requestType));
+        thread.start();
+
+        if (!(SubmitBulkPerformanceTest.threadCount < MAX_WORKER_THREADS)) {
+            try {
+                // Sleep indefinitely as thread should be interrupted to resume processing.
+                Thread.sleep(Long.MAX_VALUE);
+            } catch (InterruptedException e) {
+                // Do nothing.
+                LOGGER.debug("interrupted" + e);
+            }
+        }
+    }
+
+    /**
+     * Creates new bulk request using provided template request.
+     *
+     * @param templateBulkRequest template bulk request.
+     * @param customerRef         customer reference to be populated in new bulk request.
+     * @return BulkRequestType new instance.
+     */
+    private BulkRequestType createBulkRequest(final BulkRequestType templateBulkRequest, final String customerRef) {
+        final BulkRequestType requestType = new BulkRequestType();
+
+        HeaderType headerType = new HeaderType();
+        headerType.setCustomerReference(customerRef);
+        headerType.setRequestCount(IND_REQ_PER_BULK);
+        headerType.setSdtCustomerId(templateBulkRequest.getHeader().getSdtCustomerId());
+        headerType.setTargetApplicationId(templateBulkRequest.getHeader().getTargetApplicationId());
+        requestType.setHeader(headerType);
+
+        RequestsType requestsType = new RequestsType();
+        requestType.setRequests(requestsType);
+
+        RequestItemType template = templateBulkRequest.getRequests().getRequest().get(0);
+
+        // Create new individual requests using template.
+        for (int i = 1; i <= IND_REQ_PER_BULK; i++) {
+            RequestItemType item = cloneRequestItem(template);
+            item.setRequestId(customerRef + "-" + i);
+            requestsType.getRequest().add(item);
+        }
+
+        return requestType;
+    }
+
+    /**
+     * Clones given request item.
+     *
+     * @param source request item to clone.
+     * @return RequestItemType cloned instance.
+     */
+    private RequestItemType cloneRequestItem(final RequestItemType source) {
+        RequestItemType dest = new RequestItemType();
+        dest.setRequestId(source.getRequestId());
+        dest.setRequestType(source.getRequestType());
+        dest.setMcolClaim(source.getMcolClaim());
+        dest.setMcolWarrant(source.getMcolWarrant());
+        dest.setMcolBreathingSpace(source.getMcolBreathingSpace());
+        dest.setMcolJudgment(source.getMcolJudgment());
+        dest.setMcolClaimStatusUpdate(source.getMcolClaimStatusUpdate());
+        dest.setMcolSetAside(source.getMcolSetAside());
+        dest.setMcolJudgmentWarrant(source.getMcolJudgmentWarrant());
```

Comment:
These changes will need to be reverted once the sdtEndpointDistWsdl2Java task has been removed from the producers-api module. The changes relate to the classes generated by that task which is only intended for the sdt-consumers-sample module.

Reply:
Confirmed changes have been reverted.